src/data/polynomial/degree/definitions.lean

```diff
@@ -744,6 +744,38 @@ lemma le_nat_degree_of_coe_le_degree {P : polynomial R} {n : ℕ} (hdeg : ↑n 
   n ≤ P.nat_degree :=
 with_bot.coe_le_coe.mp ((degree_eq_nat_degree $ ne_zero_of_coe_le_degree hdeg) ▸ hdeg)
 
+lemma nat_degree_le_iff_coeff_eq_zero {f : polynomial R} (N : ℕ) :
+  f.nat_degree ≤ N ↔ ∀ n : ℕ, N < n → f.coeff n = 0 :=
+by simp_rw [nat_degree_le_iff_degree_le, degree_le_iff_coeff_zero, with_bot.coe_lt_coe]
+
+lemma nat_degree_C_mul_le (a : R) (f : polynomial R) :
+  (C a * f).nat_degree ≤ f.nat_degree :=
+calc
+  (C a * f).nat_degree ≤ (C a).nat_degree + f.nat_degree : nat_degree_mul_le
+  ... = 0 + f.nat_degree : by rw nat_degree_C a
+  ... = f.nat_degree : zero_add _
+
+lemma nat_degree_mul_C_le (f : polynomial R) (a : R) :
+  (f * C a).nat_degree ≤ f.nat_degree :=
+calc
+  (f * C a).nat_degree ≤ f.nat_degree + (C a).nat_degree : nat_degree_mul_le
+  ... = f.nat_degree + 0 : by rw nat_degree_C a
+  ... = f.nat_degree : add_zero _
+
+lemma nat_degree_mul_C_eq_of_mul_eq_one {ai : R} (au : a * ai = 1) {f : polynomial R} :
```

Comment:
These lemmas hold more generally for `\forall x, ai * a \ne 0` right? In particular, that hypothesis means you can create a primed lemma for no_zero_divisors

Reply:
I am on my phone, so I may be misunderstanding.  I agree that it suffices to assume that `a` is a non-zero divisor and I could prove it in this case, adding then a lemma saying that having an inverse implies being a non-zero divisor.  However, assuming *every* non-zero element of R is a non-zero divisor is more than I want.

Reply:
I'm suggesting both - show that it suffices for a to be a nonzero divisor, but for convenience, also provide the lemma that provides that hypothesis automatically using typeclass search when working in the reals, rationals, etc

Reply:
Eric, I decided to add a more direct lemma, assuming that the product `a * f.leading_coeff` does not vanish.  This implies quickly the result that you want about non-zero divisors.  The proofs of the results that I needed, though, have not been affected.

I also found out that the final section on `no_zero_divisors` assumed `comm_semiring` and I removed the `comm` assumption: let's see if mathlib likes this!  If it does not, I will add the hypothesis back and that refactor will wait a separate PR.